actors/evm/src/interpreter/execution.rs

```diff
@@ -181,222 +159,236 @@ impl<'r, 'a, RT: Runtime + 'r> Machine<'r, 'a, RT> {
+        SWAP9: {intrinsic}
+        SWAP10: {intrinsic}
+        SWAP11: {intrinsic}
+        SWAP12: {intrinsic}
+        SWAP13: {intrinsic}
+        SWAP14: {intrinsic}
+        SWAP15: {intrinsic}
+        SWAP16: {intrinsic}
 
         // event logs
-        LOG0: {std}
-        LOG1: {std}
-        LOG2: {std}
-        LOG3: {std}
-        LOG4: {std}
+        LOG0: {intrinsic}
+        LOG1: {intrinsic}
+        LOG2: {intrinsic}
+        LOG3: {intrinsic}
+        LOG4: {intrinsic}
 
         // create variants
-        CREATE: {std}
-        CREATE2: {std}
+        CREATE: {intrinsic}
+        CREATE2: {intrinsic}
 
         // call variants
-        CALL: {std}
-        CALLCODE: {std}
-        DELEGATECALL: {std}
-        STATICCALL: {std}
+        CALL: {intrinsic}
+        CALLCODE: {intrinsic}
+        DELEGATECALL: {intrinsic}
+        STATICCALL: {intrinsic}
 
         // control flow magic
-        JUMPDEST: {=> Ok(ControlFlow::Continue)} // noop marker opcode for valid jumps addresses
+        // noop marker opcode for valid jumps addresses
+        JUMPDEST: {(m) => {
+            m.pc += 1;
+        }}
 
         JUMP: {(m) => {
-            if let Some(dest) = instructions::JUMP(&mut m.state.stack, m.bytecode)? {
-                m.pc = dest;
-                Ok(ControlFlow::Jump)
-            } else {
-                // cant happen, unless it's a cosmic ray
-                Err(StatusCode::Failure)
-            }
+            try_ins! { JUMP(m) => (res) {
+                if let Some(dest) = res {
+                    m.pc = dest;
+                } else {
+                    // cant happen, unless it's a cosmic ray
+                    m.exit = Some(StatusCode::Failure);
+                }
+            }}
         }}
 
         JUMPI: {(m) => {
-            if let Some(dest) = instructions::JUMPI(&mut m.state.stack, m.bytecode)? {
-                m.pc = dest;
-                Ok(ControlFlow::Jump)
-            } else {
-                Ok(ControlFlow::Continue)
-            }
+            try_ins! { JUMPI(m) => (res) {
+                if let Some(dest) = res {
+                    m.pc = dest;
+                } else {
+                    m.pc += 1;
+                }
+            }}
         }}
 
         PC: {(m) => {
-            instructions::PC(&mut m.state.stack, m.pc)?;
-            Ok(ControlFlow::Continue)
+            try_ins! { PC(m) => {
+                m.pc += 1;
+            }}
         }}
 
         RETURN: {(m) => {
-            instructions::RETURN(m.state, m.system)?;
-            Ok(ControlFlow::Exit)
+            try_ins! { RETURN(m) => {
+                m.exit = Some(StatusCode::Success);
+            }}
         }}
 
         REVERT: {(m) => {
-            instructions::REVERT(m.state, m.system)?;
-            m.reverted = true;
-            Ok(ControlFlow::Exit)
+            try_ins! { REVERT(m) => {
+                m.reverted = true;
+                m.exit = Some(StatusCode::Success);
```

Comment:
future: maybe `StatusCode::Reverted`?

Reply:
If we have multiple ways to revert, yeah.